ddtrace/_trace/tracer.py

```diff
@@ -402,7 +415,7 @@ def get_log_correlation_context(self) -> Dict[str, str]:
         If there is no active span, a dictionary with an empty string for each value will be returned.
         """
         active: Optional[Union[Context, Span]] = None
-        if self.enabled:
+        if self.enabled or self._asm_enabled:
```

Comment:
Counterintuitively, this approach couples `Tracer` to the ASM product. You might consider adding a layer of abstraction that makes `Tracer` hold a generic list of "products". This might be too much of a scope expansion for your work at the moment, so I'd probably still approve this approach. In the long run, though, the `Tracer` mustn't know specifics about other products.

Reply:
Would this be better?

```suggestion
        if self.enabled or self._apm_opt_out:
```